### PR TITLE
Conditional InternalTransition

### PR DESCRIPTION
**Pull request by Tuan-Minh**
Hi, I have add one InternalTransitionIf for each InternalTransition methods, nothing special about them to write about.

### CODE REVIEW

**Comment by HenningNT** [comment]
This looks very good! I haven't tried it yet, but it looks like it can be merged as is.

File: src/Stateless/InternalTriggerBehaviour.cs [hunk 1]
```diff
@@ -10,6 +10,10 @@ public InternalTriggerBehaviour(TTrigger trigger)
                 : base(trigger, new TransitionGuard(() => true, "Internal Transition"))
             {
             }
+            public InternalTriggerBehaviour(TTrigger trigger, Func<bool> guard)
+                : base(trigger, new TransitionGuard(guard, "Internal Transition"))
+            {
+            }
             public override bool ResultsInTransitionFrom(TState source, object[] args, out TState destination)
```
Comment (nblumhardt) [inline src/Stateless/InternalTriggerBehaviour.cs]: Missing newlines between the method bodies here.

File: src/Stateless/InternalTriggerBehaviour.cs [hunk 2]
```diff
@@ -10,6 +10,10 @@ public InternalTriggerBehaviour(TTrigger trigger)
                 : base(trigger, new TransitionGuard(() => true, "Internal Transition"))
```
Comment (nblumhardt) [inline src/Stateless/InternalTriggerBehaviour.cs]: Chaining constructors would keep things DRYer here:

```
    :this(trigger, () => true)
```

**Comment by nblumhardt** [comment]
Great PR, thanks!

It would cut the amount of duplication in the code quite a bit if we redirected the old methods (without `guard`s) to call the new methods. I made a comment regarding the constructor of `InternalTriggerBehavior`, but I think the same applies to `InternalPermitIf()` and so-on.

**Comment by Tuan-Minh** [comment]
@nblumhardt I have address your comment regarding duplicitous code. Silly me!

**Comment by nblumhardt** [comment]
Great! Thanks.

**Pull request merged**